src/compiler-attrs.md

```diff
@@ -0,0 +1,18 @@
+# Compiler attributes
+
+The following [attributes] affect internal settings of the compiler.
+
+<!-- TODO: feature -->
```

Comment:
I've been thinking about `feature` while sleeping, and think that while we should include it in the index, we should do so as a note and point to the rustc book. We should otherwise not discuss it since it will never be a part of stable Rust.